### PR TITLE
Add `to_XXXpoint()` unit tests. Fix exception clause

### PR DESCRIPTION
**Pull request by bpintea**
This adds unit tests for `to_geopoint()` and `to_cartesianpoint()` functions.

It also corrects the `SpatialPoint` string parsing to only throw `IllegalArgumentException`, which is the only exception handled in the from-string evaluator.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java [hunk 1]
```diff
@@ -58,7 +64,7 @@ protected NodeInfo<? extends Expression> info() {
         return NodeInfo.create(this, ToGeoPoint::new, field());
     }
 
-    @ConvertEvaluator(extraName = "FromString", warnExceptions = { IllegalArgumentException.class })
+    @ConvertEvaluator(extraName = "FromString", warnExceptions = { RuntimeException.class })
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java:67]: Is it not better to change the parser to throw IllegalArgumentException? I did that in my other PR (reading from source), but we could view this as a bug-fix for 8.12 and do it there.

Comment (craigtaverner) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java:67]: See https://github.com/elastic/elasticsearch/pull/102880/files#diff-273e58854c0ce3de04bf9c2b0c750d1db8549254cda977f766e46e5d3244a5b6R81

**Review by craigtaverner — CHANGES_REQUESTED** [review]
Great to add these tests, but I think we should not use RuntimeException.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java (hunk 1 above)
Comment (craigtaverner) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java]: Done in 35cbe5a19676fd560665769b262279d4e01e8cfa

**Review by craigtaverner — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java]: > Is it not better to change the parser to throw IllegalArgumentException?

It is, IMO too, but wasn't sure why the `RuntimeException` was used in the 1st place, so adapted around it.

**Pull request merged**